### PR TITLE
Add validation for disk_format

### PR DESCRIPTION
**Pull request by jschmid1**
closes https://github.com/SUSE/DeepSea/issues/436

### CODE REVIEW

**Comment by swiftgist** [comment]
Won't this prevent different minions from having different formats?

**Comment by jschmid1** [comment]
True, we should allow this.
Updating..

File: srv/modules/runners/validate.py [hunk 1]
```diff
@@ -283,6 +283,30 @@ def mgrs(self):
...
+	"""
+        formats = {}
+        for node in self.data.keys():
+            # we can detect disk-formats only in the new yaml format
+            if ('ceph' in self.data[node] and
+                'storage' in self.data[node]['ceph']):
+                devices = self.data[node]['ceph']['storage']['osds']
+                formats[node] = []
+                for dev_name in devices.keys():
+                    if 'format' in devices[dev_name]:
```
Comment (jan--f) [inline srv/modules/runners/validate.py]: if ```format``` is not a key in the profile _I think_ we default to bluestore. Maybe we should add an else branch here?

File: srv/modules/runners/validate.py [hunk 2]
```diff
@@ -283,6 +283,30 @@ def mgrs(self):
...
+                       formats[node].append(devices[dev_name]['format'])
+        for node, df in formats.iteritems():
+	    if len(set(formats)) > 1:
```
Comment (jan--f) [inline srv/modules/runners/validate.py]: Is ```formats``` correct here? Shouldn't ```len(set(df))``` be checked?

File: srv/modules/runners/validate.py [hunk 3]
```diff
@@ -283,6 +283,30 @@ def mgrs(self):
...
+		msg = "Please use uniform Disk Formats across nodes. Either Bluestore or Filestore"
```
Comment (jan--f) [inline srv/modules/runners/validate.py]: To me this sounds like "One format across all nodes". How about ```Ceph only supports on OSD backend per host. Please use either Filestore or Bluestore for OSDs on any one host.```

Comment (jan--f) [inline srv/modules/runners/validate.py]: @smithfarm Could you lend your professional/nativ-speaker eye please?

**Pull request closed without merging**